### PR TITLE
[#2979] fix problem with generating OSGI manifests

### PR DESCRIPTION
**Pull request by jherkel**
I had to change some things in master pom : 
1) maven-checkstyle-plugin - I had a problem with a generated class within querydsl-osgi-itest-jpa maven subproject
2) maven-javadoc-plugin - I needed to append javax.interceptor/javax.interceptor-api because I  had a problem with generating javadocs with JDK8 when @Transactional annotation was used
3) Create additional 'osgi' profile

### CODE REVIEW

**Review by F43nd1r — APPROVED** [review]
Looks good to me, but please refrain from commiting unrelated whitespace changes, makes it harder to read the diff.

File: pom.xml [hunk 1]
```diff
@@ -74,9 +76,7 @@
     <dokka.version>1.5.0</dokka.version>
 
     <!-- Import-Package definitions for maven-bundle-plugin -->
-    <osgi.import.package.root>
-      *
-    </osgi.import.package.root>
+    <osgi.import.package.root>*</osgi.import.package.root>
```
Comment (jwgmeligmeyling) [inline pom.xml:79]: Isn't `com.querydsl.*` a more sensible default then `*` here?

File: pom.xml [hunk 2]
```diff
@@ -819,6 +836,16 @@
         <module>querydsl-maven-plugin</module>
       </modules>
     </profile>
+    
+    <profile>
+      <id>osgi</id>
```
Comment (jwgmeligmeyling) [inline pom.xml:841]: What is the purpose of this `osgi` module?

File: querydsl-jdo/pom.xml [hunk 3]
```diff
@@ -146,6 +146,7 @@
         </executions>
         <configuration>
           <archive>
+            <manifestFile>${project.build.outputDirectory}/META-INF/MANIFEST.MF</manifestFile>
```
Comment (jwgmeligmeyling) [inline querydsl-jdo/pom.xml:149]: Are manifest entries (below) still added if a manifest file is provided?

**Review by jwgmeligmeyling — COMMENTED** [review]
I have a few questions, besides that the changes look good.

Thanks for adding an example case as well, this will really help us to maintain OSGi support down the road!

File: pom.xml (hunk 1 above)
Comment (jherkel) [reply pom.xml:79]: no, '*' is default for bnd and it means that bnd imports all referred packages

File: querydsl-jdo/pom.xml (hunk 3 above)
Comment (jherkel) [reply querydsl-jdo/pom.xml:149]: yes, you can check that final jar contains also Automatic-Module-Name. These missing OSGI headers can be fixed also that we change <packaging> from jar to bundle. (bundle also create a jar archive)

File: pom.xml (hunk 2 above)
Comment (jherkel) [reply pom.xml:841]: I wanted to select out these integration tests for OSGI.

**Comment by F43nd1r** [comment]
@jwgmeligmeyling questions seem answered, is this good to merge?

**Comment by jwgmeligmeyling** [comment]
I reviewed it, but didn't have the change to test it (package and compare the manifests for each of the modules).

**Pull request closed without merging**